avocado/core/runners/requirement_package.py

```diff
@@ -0,0 +1,58 @@
+import os
+import subprocess
+import time
+
+from ...utils import exit_codes
+from .. import nrunner
+
+
+class RequirementPackageRunner(nrunner.BaseRunner):
+    """
+    Runner for requirements of type package
+
+    This runner handles, initially, the installation of packages using the
+    avocado-software-manager. It can be easily extended to support other
+    arguments, like `remove`.
+
+    The arguments represent the packages that should be installed.
```

Comment:
> This would make you think of the usage of the runnable attributes, such as:
> 
> * Should the `uri` be used for any purpose?

I was thinking about the arguments while I was developing it and I could not find a way to give a mean to the `uri` argument. The executable/binary will be the runner, detected automatically by the runnable. That is why it is not passed to this current implementation.

> * Does it make sense to use `kwargs` as environment variables for this runner?

Right now I don't see a use case for that. This does not mean we cannot evolve it later.

> * How would we implement a `check if installed` action, in addition to the `install` one?

The only reason I see to check if something is installed without installing it is when there is no network connectivity. I was thinking about following the same pattern in the `tap` resolver and have the first argument as a command and the next
argument as the package. This allows the extension to `list` and `remove` without complex code.

Reply:
> > This would make you think of the usage of the runnable attributes, such as:
> > 
> > * Should the `uri` be used for any purpose?
> 
> I was thinking about the arguments while I was developing it and I could not find a way to give a mean to the `uri` argument. The executable/binary will be the runner, detected automatically by the runnable. That is why it is not passed to this current implementation.

AFAICT no requirement is currently created with more than one package name, right?  At the same time, a requirement with multiple package names seems like a worthy feature.  Because of that, I can not think of something like a package name mapping perfectly to the URI, or any other aspect of this requirement type that would.  And the lack of a URI may be fine.  I've checked, and this works perfectly fine:

```
$ avocado-runner-requirement-package runnable-run -a package_name
``` 

But, let's document that the URI is not being used for this runner.

> > * Does it make sense to use `kwargs` as environment variables for this runner?
> 
> Right now I don't see a use case for that. This does not mean we cannot evolve it later.
> 

If there's no case for it now, it should be removed.  We should not introduce code whose behavior/purpose can not be justified at the time it's being proposed. 

I've tested the `avocado-runner-requirement-package` without the environment variables handling and there's no impact to its behavior.

> > * How would we implement a `check if installed` action, in addition to the `install` one?
> 
> The only reason I see to check if something is installed without installing it is when there is no network connectivity. I was thinking about following the same pattern in the `tap` resolver and have the first argument as a command and the next
> argument as the package. This allows the extension to `list` and `remove` without complex code.

I think giving special meaning to the "args" (aka `-a` command line arguments) based on their position can get very confusing and error prone.  Having parameters that are not known to the basic `Runnable` data structure and `runnable-run` command line is the point of the `kwargs` (aka `KEY=VALUE`) arguments.

So, I'd go with having something like:

```
     Runnable attributes usage:

     * uri: not used

     * args: package names to act upon

     * kwargs: the following keyword based arguments are recognized:
       - action: one of install, check, $others . Defaults to `install` if no explicit one given. 
```

On top of that, we *can*, in the future, evolve a common interface across requirements.  BTW, a generic `check` requirement can be *very* useful when fulfilling the requirement is either too complex or impossible. Still, it can serve as a generic way of not running tests that are missing requirements.